cli/command/stack/deploy_composefile.go

```diff
@@ -320,6 +322,13 @@ func deployServices(
 			if sendAuth {
 				updateOpts.EncodedRegistryAuth = encodedAuth
 			}
+
+			if image != service.Spec.TaskTemplate.ContainerSpec.Image {
```

Comment:
The label can give us the same signal as the `--image` flag on service commands.

Here's how it works:
* in `convert.Services()` we add a label to each service with a key of `com.docker.stack.image` (or something like this) and a value of the literal string from the compose file (`composetypes.Service.Image`)
* when we enter this `deployServices()` code we compare the existing label value against the new value from the compose file

```go
image := serviceSpec.TaskTemplate.ContainerSpec.Image
// has the image value changed in the compose file?
if service.Spec.Labels["com.docker.stack.image"] != image {
    if !noResolveImage {
        updateOpts.QueryRegistry = true
    }
}
```

This is different from the current code because we don't know if `service.Spec.TaskTemplate.ContainerSpec.Image` was modified or not, but we do know the label is never modified by other code.

Does that make sense?